File: finance-calc/pkg/scenario/models/scenario.go
package scenario

import (
	debtModels "financeCalc/pkg/debt/models"
	savingsModels "financeCalc/pkg/savings/models"
	"time"
)

type Scenario struct {
	Id              int                             `json:"id"`
	StartDate       time.Time                       `json:"startDate"`
	Debts           []*debtModels.Debt              `json:"debts"`
	SavingsAccounts []*savingsModels.SavingsAccount `json:"savings"`
	SortKey         string                          `json:"sortKey"`
	ReverseSort     bool                            `json:"reverseSort"`
}

File: webapp/src/types.d/scenario/scenario.ts
import { Debt } from "./debt";
import { SavingsAccount } from "./savings-account";
import { SortKey } from "../sort-keys";

export interface Scenario {
  startDate?: string;
  debts: Array<Debt>;
  savingsAccounts: Array<SavingsAccount>;
  sortKey: string;
  reverse: boolean;
}

